src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlArrayTranslator.cs

```diff
@@ -33,6 +33,12 @@ public class NpgsqlArrayTranslator : IMethodCallTranslator, IMemberTranslator
             typeof(Enumerable).GetTypeInfo().GetMethods(BindingFlags.Public | BindingFlags.Static | BindingFlags.DeclaredOnly)
                 .Single(mi => mi.Name == nameof(Enumerable.Any) && mi.GetParameters().Length == 1);
 
+        [NotNull] static readonly MethodInfo StringArrayJoin =
+            typeof(EnumerableExtensions).GetTypeInfo()
+                .GetMethods(BindingFlags.Public | BindingFlags.Static | BindingFlags.DeclaredOnly).Single(mi =>
+                    mi.Name == nameof(EnumerableExtensions.Join) &&
+                    mi.GetParameters()[1].ParameterType == typeof(string));
```

Comment:
It would be better to write this using the existing style as above.
```suggestion
            typeof(EnumerableExtensions).GetTypeInfo().GetMethods(BindingFlags.Public | BindingFlags.Static | BindingFlags.DeclaredOnly)
                .Single(mi => mi.Name == nameof(EnumerableExtensions.Join) && mi.GetParameters()[1].ParameterType == typeof(string));
```

By the way, this code could fail if someone add `Join` accepting only one parameter. Therefore, you should check parameter count too. Use `ElementAtOrDefault` to get the parameter.